tests/unit/test_charm.py

```diff
@@ -697,3 +697,8 @@ def test_on_remove_sysctl_is_deleted(harness: Harness):
         harness.charm.on.remove.emit()
 
         patched_sysctl_remove.assert_called_once()
+
+
+def test_workload_version(harness, monkeypatch):
+    monkeypatch.setattr(harness.charm.workload, "get_version", Mock(return_value="1.2.3"))
+    assert harness.charm.version == "1.2.3"
```

Comment:
**question:** What is this testing? From what I can see, we patch a property to return a fixed value, then assert that fixed value.

**suggestion:** I think a few better tests might be to:
1. Patch `run_bin_command` to give a fixed output that we copied from a Kafka log, and check that we return `""` or `$VERSION` successfully
2. Test that after running the `install` hook, `harness.charm.unit.workload_version` (does that attr exist?), we get a value
3. Same as 2, but after running `config-changed` hook

Reply:
I guess my confusion came from the fact that this looks more like a scenario test rather than a unit test, but they are the same thing in this code base. Anyway, thanks for the suggestion, please let me know what you think about this new proposal